pkg/controller/datavolume/pvc-clone-controller.go

```diff
@@ -642,6 +645,15 @@ func (r *PvcCloneReconciler) makeSizeDetectionPodSpec(
 	if err != nil {
 		return nil
 	}
+	if l5d.IsMeshed(sourcePvc.GetAnnotations()) {
+		container.Lifecycle = &v1.Lifecycle{
+			PreStop: l5d.PreStopHook(),
+		}
+	} else if istio.IsMeshed(sourcePvc.GetAnnotations()) {
+		container.Lifecycle = &v1.Lifecycle{
+			PreStop: istio.PreStopHook(),
+		}
+	}
```

Comment:
instead of doing this everywhere maybe have a helper function in the `mesh` package that takes a pod and adds the hooks if appropriate?